crates/electrum/src/electrum_ext.rs

```diff
@@ -191,38 +186,26 @@ pub trait ElectrumExt<A> {
             .enumerate()
             .map(|(i, spk)| (i as u32, spk));
 
-        self.scan(
-            prev_tip,
-            [((), spk_iter)].into(),
-            txids,
-            outpoints,
-            usize::MAX,
-            batch_size,
-        )
+        self.sync_with_keychain([((), spk_iter)].into())
     }
 }
 
 impl ElectrumExt<ConfirmationHeightAnchor> for Client {
-    fn scan<K: Ord + Clone>(
+    fn sync_with_keychain<K: Ord + Clone>(
         &self,
-        prev_tip: Option<CheckPoint>,
         keychain_spks: BTreeMap<K, impl IntoIterator<Item = (u32, ScriptBuf)>>,
-        txids: impl IntoIterator<Item = Txid>,
-        outpoints: impl IntoIterator<Item = OutPoint>,
-        stop_gap: usize,
-        batch_size: usize,
     ) -> Result<ElectrumUpdate<K, ConfirmationHeightAnchor>, Error> {
         let mut request_spks = keychain_spks
             .into_iter()
             .map(|(k, s)| (k, s.into_iter()))
             .collect::<BTreeMap<K, _>>();
         let mut scanned_spks = BTreeMap::<(K, u32), (ScriptBuf, bool)>::new();
 
-        let txids = txids.into_iter().collect::<Vec<_>>();
-        let outpoints = outpoints.into_iter().collect::<Vec<_>>();
+        let txids = Vec::<Txid>::new();
```

Comment:
`sync_with_keychain` no longer has these fields, Should they remain empty `Vec`s or is there more refactor needed @vladimirfomene @LLFourn

Reply:
Vec works fine here